.changelog/add-change.sh

```diff
@@ -0,0 +1,377 @@
+    is provided using flags, it is the <id>. If neither are provided using
+    flags, it is the <section> only if it is a valid section match, otherwise it
+    is the <id>. It is an error if there is one of these, but both the
+    <id> and <section> were provided using flags.
+
+EOF
+
+}
+
+messages=()
+
+while [[ "$#" -gt '0' ]]; do
+    case "$1" in
+        -h|--help)
+            show_usage
+            exit 0
+            ;;
+        -v|--verbose)
+            verbose="$1"
+            ;;
+        -n|--issue|--issue-no)
+            if [[ -z "$2" ]]; then
+                printf 'No argument provided after %s\n' "$1"
+                exit 1
+            fi
+            num="$2"
+            num_type='issue'
+            shift
+            ;;
+        -p|--pr|--pull-request)
+            if [[ -z "$2" ]]; then
+                printf 'No argument provided after %s\n' "$1"
+                exit 1
+            fi
+            num="$2"
+            num_type='pr'
+            shift
+            ;;
+        -i|--id)
+            if [[ -z "$2" ]]; then
+                printf 'No argument provided after %s\n' "$1"
+                exit 1
+            fi
+            id="$2"
+            shift
+            ;;
+        -s|--section)
+            if [[ -z "$2" ]]; then
+                printf 'No argument provided after %s\n' "$1"
+                exit 1
+            fi
+            section="$2"
+            shift
+            ;;
+        -m|--message)
+            if [[ -z "$2" ]]; then
+                printf 'No argument provided after %s\n' "$1"
+                exit 1
+            fi
+            messages+=( "$2" )
+            shift
+            ;;
+        *)
+            if [[ "$1" =~ [[:space:]] ]]; then
+                messages+=( "$1" )
+            elif [[ "$1" =~ ^[[:digit:]]+$ ]]; then
+                if [[ -n "$num" ]]; then
+                    printf 'Unknown argument: [%s]. The <num> was already provided as [%s] using a flag.\n' "$1", "$num"
+                    exit 1
+                fi
+                num="$1"
+                num_type='pr'
+            elif [[ "$1" =~ ^[-[:alnum:]]+$ ]]; then
+                if [[ -z "$id_sect_1" ]]; then
+                    id_sect_1="$1"
+                elif [[ -z "$id_sect_2" ]]; then
+                    id_sect_2="$1"
+                else
+                    printf 'Unknown argument: [%s]. An <id> and <section> were already provided.\n' "$1"
+                    exit
+                fi
+            else
+                printf 'Unknown argument: [%s].\n' "$1"
+                printf 'The <num> can only contain digits.\n'
+                printf 'The <id> and <section> can only contain alphanumeric characters and dashes.\n'
+                printf 'The <message> must contain at least one space.\n'
+                exit 1
+            fi
+            ;;
+    esac
+    shift
+done
+
+where_i_am="$( cd "$( dirname "${BASH_SOURCE:-$0}" )"; pwd -P )"
```

Comment:
Good catch. Fixed.

Reply:
Sorry. Github pulled a switcharoo on me and my reply was added to this one instead of the one I meant to.

This line is fine. The only way it'll fail is if the `dirname` utility isn't available, but it's a standard utility, so it's safe to assume it's there.

Also, you put the `exit` in the wrong spot in your suggestion. Where you have it just exits the sub-process.